openmdao/drivers/tests/test_scipy_optimizer.py

```diff
@@ -985,6 +996,52 @@ def __init__(self, problem):
                 self.problem = problem
                 self.counter = 0
 
+            def setup(self):
+                self.add_input('x', np.array([1.5, 1.5, 1.5]))
+                self.add_output('f', 0.0)
+                self.declare_partials('f', 'x', method='fd', form='central', step=1e-4)
+
+            def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+                x = inputs['x']
+                outputs['f'] = rosenbrock(x)
+
+        x0 = np.array([0.5, 0.8, 1.4])
+
+        prob = Problem()
+        model = prob.model
+        indeps = prob.model.add_subsystem('indeps', IndepVarComp(problem=prob))
+        indeps.add_output('x', list(x0))
+
+        model.add_subsystem('rosen', Rosenbrock(problem=prob))
+        model.add_subsystem('con', ExecComp('c=sum(x)', x=np.ones(3)))
+        model.connect('indeps.x', 'rosen.x')
+        model.connect('indeps.x', 'con.x')
+        prob.driver = driver = ScipyOptimizeDriver()
+        driver.options['optimizer'] = 'trust-constr'
+        driver.options['tol'] = 1e-5
+        driver.options['maxiter'] = 2000
+        driver.options['disp'] = False
+
+        model.add_design_var('indeps.x')
+        model.add_objective('rosen.f', scaler=1/rosenbrock(x0))
+        model.add_constraint('con.c', equals=1.)
+
+        prob.setup()
+        prob.run_driver()
+
+        assert_rel_error(self, prob['con.c'], 1., 1e-3)
+
+    @unittest.skipUnless(LooseVersion(scipy_version) >= LooseVersion("1.1"),
+                         "scipy >= 1.1 is required.")
+    def test_trust_constr_inequality_con(self):
+
+        class Sphere(ExplicitComponent):
+
+            def __init__(self, problem):
+                super(Sphere, self).__init__()
+                self.problem = problem
```

Comment:
There are multiple instances in this file where the Problem is passed as an __init__ arg to a Component, e.g. Sphere and Rosenbrock.  As far as I can tell, the problem attribute of these components is never used so we should remove the __init__ arg and the attribute anywhere where it currently appears in the file.